ensime_shared/ensime.py

```diff
@@ -279,11 +336,15 @@ def au_buf_leave(self, client, filename):
 
     @execute_with_client()
     def au_cursor_hold(self, client, filename):
-        client.on_cursor_hold(filename)
+        """Handler for CursorHold autocommand event."""
+        self.watchdog_client_server_connection(client)
+        client.unqueue_and_display(filename)
+        client.editor.cursorhold()
```

Comment:
Deoplete while using `omnifunc` does wait for the response in a synchronous way, and more often than I'd like ensime server gets really really slow to respond having to wait several seconds after each keystroke what makes a terrible typing experience, that's why I want to implemente a deoplete source so the integration with autocomplete will be asynchronous avoiding those horrible blocks. But we have to work on ensime-vim integration API first.